src/State/Provider/DeserializeProvider.php

```diff
@@ -75,7 +75,7 @@ public function provide(Operation $operation, array $uriVariables = [], array $c
             throw new UnsupportedMediaTypeHttpException('Format not supported.');
         }
 
-        if ($operation instanceof HttpOperation && null === ($serializerContext[SerializerContextBuilderInterface::ASSIGN_OBJECT_TO_POPULATE] ?? null)) {
```

Comment:
There is
```
if (!$operation instanceof HttpOperation || !($request = $context['request'] ?? null)) {
            return $this->decorated?->provide($operation, $uriVariables, $context);
        }
```
before

Reply:
I've also improved HttpOperation detection recently with https://github.com/api-platform/core/commit/4dc91a0d0367351322fe6fed1703d0de19e5deb9 don't hesitate if you find things that needs type definition improvement (we can always track them in a separated issue)

Reply:
There is another PHPStan failure to fix since the recent symfony release
https://github.com/api-platform/core/pull/7255